lib/puppet/type/iis_site.rb

```diff
@@ -171,6 +171,10 @@ def insync?(is)
       end
       value
     end
+
+    def insync?(is)
+      is.sort_by { |a| ((a['protocol'] == 'https') ? '0' : '1') + a['bindinginformation'] } == should.sort_by { |a| ((a['protocol'] == 'https') ? '0' : '1') + a['bindinginformation'] }
```

Comment:
Given the sorting code is used here and in the provider, would it be better to move that to into a module method in the puppet_x namespace e.g.  `puppetlabs-iis/lib/puppet_x/puppetlabs/iis/bindings.rb`

Reply:
I confess that I'm woefully ignorant about ruby code organization.

Reply:
It's also a little dangerous but that's a Puppet thing.  The type here is depending on the Provider to have the `sort_by` method available.  And you've only implemented that on the WebAdministration provider. If this used the IISAdministration provider it would fail with method not found.

Reply:
Trying it now...